backend/src/app.py

```diff
@@ -18,56 +21,99 @@
 
 @app.route("/")
 def index():
+    user_id = request.cookies.get("user_id")
+    access_token = request.cookies.get("spotify_access_token")
     limit = request.args.get("limit")
     offset = request.args.get("offset")
-    playlists = spotify.get_playlists(limit=limit, offset=offset)
+    playlists = spotify.get_playlists(
+        user_id=user_id, access_token=access_token, limit=limit, offset=offset
+    )
     sort_by = request.args.get("sort_by")
     desc = request.args.get("desc") == "True"
     if sort_by is not None:
         playlists.sort(key=lambda x: x[sort_by], reverse=desc)
-    return playlists
+    return [playlist.model_dump() for playlist in playlists]
+
+
+@app.route("/auth/login")
+def login():
+    state = "thisShouldBeARandomString"
+    query_string = spotify.get_login_query_string(state)
+    return "https://accounts.spotify.com/authorize?" + query_string
```

Comment:
In case you're unsure (as opposed to just taking a short cut for now) - typically for this state we'll:
* Generate a random string, as you say
* Set it on the users session
* Then when we get to the `get-user-code` endpoint, we check the state we were sent matches that in session

Reply:
Ah yes, good point. Think I did this just to get it working then forgot about it. I'll see if I can get that fixed.

Reply:
I think we run into an issue here. Flask sessions only seem to work for flask frontends. I can't seem to access session variables across requests. I'm going to update this to a uuid but still not check it in the `auth/get-user-code` function for now. If I have some time before we get to week 10 I'll look at some way of persisting session info across requests.

Reply:
Hmm, I'd be surprised if that was strictly true as the backend shouldn't know/care who exactly sent a request, but we might have some cookie wrangling to achieve as session gets stored in cookies by default, and I can certainly believe that we need to modify how our frontend handles cookie set/send behaviour